File: create-dirs.py
import os
from pathlib import Path

base_dir = r'c:\MY CODE\my codingan\Web Barber'
dirs = [
    'src',
    'src/components',
    'src/components/ui',
    'src/components/layout',
    'src/components/sections',
    'src/constants',
    'src/utils',
    'src/hooks',
    'public'
]

print(f"Creating directories in: {base_dir}\n")

for dir_path in dirs:
    full_path = os.path.join(base_dir, dir_path)
    Path(full_path).mkdir(parents=True, exist_ok=True)
    print(f"✓ Created: {dir_path}")

print('\nDirectory structure created successfully!\n')
print('Verifying directory structure:')
print('================================\n')

def list_dirs(directory, indent=''):
    items = sorted(os.listdir(directory))
    for item in items:
        full_path = os.path.join(directory, item)
        if os.path.isdir(full_path):
            print(f"{indent}📁 {item}/")
            list_dirs(full_path, indent + '  ')

list_dirs(base_dir)

File: create-dirs.js
const fs = require("fs");
const path = require("path");

const baseDir = path.resolve(__dirname);
const dirs = [
  "src",
  "src/components",
  "src/components/ui",
  "src/components/layout",
  "src/components/sections",
  "src/constants",
  "src/utils",
  "src/hooks",
  "public",
];

console.log(`Creating directories in: ${baseDir}\n`);

dirs.forEach((dir) => {
  const fullPath = path.join(baseDir, dir);
  if (!fs.existsSync(fullPath)) {
    fs.mkdirSync(fullPath, { recursive: true });
    console.log(`✓ Created: ${dir}`);
  } else {
    console.log(`✓ Already exists: ${dir}`);
  }
});

console.log("\nDirectory structure created successfully!\n");
console.log("Verifying directory structure:");
console.log("================================\n");

function listDirs(dir, indent = "") {
  const items = fs.readdirSync(dir).sort();
  items.forEach((item) => {
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      console.log(`${indent}📁 ${item}/`);
      listDirs(fullPath, indent + "  ");
    }
  });
}

listDirs(baseDir);
